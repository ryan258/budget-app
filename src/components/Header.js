import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <nav>
      <ul>
        <li>
          <NavLink to="/" exact={true} activeClassName="is-active">
            Home
          </NavLink>
        </li>
        <li>
          <NavLink to="/create" activeClassName="is-active">
            Create Expense
          </NavLink>
        </li>
      </ul>
    </nav>
  </header>
)

export default Header
