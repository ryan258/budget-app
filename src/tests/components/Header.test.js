// we need a way to virtually render our component
// so we apply snapshot testing with the react-test-renderer library
// - which allows us to render our components in regular JS code
// - then we can assert something about what got rendered
// there are 2 mains ways to test components
// - shallow rendering - not worried about user interaction or life cycle events - just concerned with what is getting rendered
// - full dom rendering - renders child components
// SNAPSHOTS - allow us to track changes to data over time and we'll get notified if it ever changes
// - jest has a handy method for this - .toMatchSnapshot()
import React from 'react'
import { shallow } from 'enzyme'

import Header from '../../components/Header'

test('should render Header correctly', () => {
  // pass in the JSX we want to shallow render
  const wrapper = shallow(<Header />)
  expect(wrapper).toMatchSnapshot()

  // now we have access to the full api to apply to our wrapper
  // .find() allows us to select areas inside our component and make assertions of them - it's like querySelectorAll
  // expect(wrapper.find('h1').length).toBe(1)
  // expect(wrapper.find('h1').text()).toBe('Expensify')
})

//~---------------- OLD react-test-renderer way, now begins the dawn of Enyme!
// test('should render Header correctly', () => {
//   //! 1) Create a new renderer
//   const renderer = new ReactShallowRenderer()
//   //! 2) Render the JSX of what we're trying to render
//   renderer.render(<Header />)
//   //! 3) assert rendered output
//   // - first time you run this test case it will always pass bc there is no existing snapshot, so jest will create a new one of the rendered header output
//   // - second time we run this case it will compare that one with the first one, then match = pass, no-match = fail
//   expect(renderer.getRenderOutput()).toMatchSnapshot()
//   console.log(renderer.getRenderOutput())
// })
