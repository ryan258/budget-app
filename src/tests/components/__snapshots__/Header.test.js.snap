// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <nav>
    <ul>
      <li>
        <NavLink
          activeClassName="is-active"
          aria-current="page"
          exact={true}
          to="/"
        >
          Home
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="is-active"
          aria-current="page"
          to="/create"
        >
          Create Expense
        </NavLink>
      </li>
    </ul>
  </nav>
</header>
`;
